sdk/jobs/pipelines/1h_automl_in_pipeline/automl-image-object-detection-in-pipeline.ipynb

```diff
@@ -0,0 +1,449 @@
+    "                else:\n",
+    "                    # train annotation\n",
+    "                    train_f.write(json.dumps(json_line) + \"\\n\")\n",
+    "            else:\n",
+    "                print(\"Skipping unknown file: {}\".format(filename))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Basic pipeline job with Image Object Detection task\n",
+    "\n",
+    "## 2.1 Build pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# note that the used docker image doesn't suit for all size of gpu compute. Please use the following command to create gpu compute if experiment failed\n",
+    "# !az ml compute create -n gpu-cluster --type amlcompute --min-instances 0 --max-instances 4 --size Standard_NC12"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define pipeline\n",
+    "@pipeline(\n",
+    "    description=\"AutoML Image Object Detection Pipeline\",\n",
+    "    default_compute=\"gpu-cluster\",\n",
+    ")\n",
+    "def automl_image_object_detection(\n",
+    "    image_object_detection_train_data,\n",
+    "    image_object_detection_validation_data\n",
+    "):\n",
+    "    # define the automl image_object-detection task with automl function\n",
+    "    image_object_detection_node = image_object_detection(\n",
+    "        training_data=image_object_detection_train_data,\n",
+    "        validation_data=image_object_detection_validation_data,\n",
+    "        target_column_name=\"label\",\n",
+    "        primary_metric=\"MeanAveragePrecision\",\n",
+    "        properties={\"_automl_internal_label\": \"latest\",},\n",
+    "        # currently need to specify outputs \"mlflow_model\" explictly to reference it in following nodes \n",
+    "        outputs={\"best_model\": Output(type=\"mlflow_model\")},\n",
+    "    )\n",
+    "    image_object_detection_node.set_limits(timeout_minutes=180)\n",
+    "\n",
+    "    image_object_detection_node.extend_search_space(\n",
+    "        [\n",
+    "            ImageObjectDetectionSearchSpace(\n",
+    "                model_name=Choice([\"yolov5\"]),\n",
+    "                learning_rate=Uniform(0.0001, 0.01),\n",
+    "                model_size=Choice([\"small\", \"medium\"]),  # model-specific\n",
+    "            ),\n",
+    "            ImageObjectDetectionSearchSpace(\n",
+    "                model_name=Choice([\"fasterrcnn_resnet50_fpn\"]),\n",
+    "                learning_rate=Uniform(0.0001, 0.001),\n",
+    "                optimizer=Choice([\"sgd\", \"adam\", \"adamw\"]),\n",
+    "                min_size=Choice([600, 800]),  # model-specific\n",
+    "            ),\n",
+    "        ]\n",
+    "    )\n",
+    "    \n",
+    "    image_object_detection_node.set_image_model(nms_iou_threshold=0.7)\n",
```

Comment:
Any reason for using this parameter in the image model ? If possible, please use the same settings as in cli examples.